app/server.go

```diff
@@ -747,33 +750,37 @@ func doSessionCleanup(s *Server) {
 	s.Store.Session().Cleanup(model.GetMillis(), SESSIONS_CLEANUP_BATCH_SIZE)
 }
 
-func (s *Server) StartElasticsearch() {
-	s.Go(func() {
-		if err := s.Elasticsearch.Start(); err != nil {
-			s.Log.Error(err.Error())
-		}
-	})
+func (s *Server) StartSearchEngine() {
+	if s.SearchEngine.ElasticsearchEngine != nil && s.SearchEngine.ElasticsearchEngine.IsActive() {
```

Comment:
We are enabling all the search engines, not `the` active one, because we can have multiple active search engines, so the idea here is to start all of that needs any start process and are active.

Reply:
Ok noted. On the other hand I was thinking to extract this out but I don't think we're going to have more than 2 or 3 search engines so this is not going to bloat the things up here

Reply:
One question, what is the purpose of having different search engines?

Reply:
The main one is fallback mechanism, but you can relay in multiple backends for different things, for example, use autocompletion from one backend and post search from another.

The idea is, if you have enabled multiple backends, you index the things in all of them, and then you try to reply to the queries in the order of priority (1. ElasticSearch, 2. Bleve, 3. Database). If one of them is not responding, or not enable for that specific search, jumps to the next one.